test/e2e/webhook_e2e_test.go

```diff
@@ -31,43 +31,42 @@ var _ = Describe("CSVs with a Webhook", func() {
 	var crc versioned.Interface
 	var namespace *corev1.Namespace
 	var nsCleanupFunc cleanupFunc
+	var nsLabels map[string]string
 	BeforeEach(func() {
 		c = newKubeClient(GinkgoT())
 		crc = newCRClient(GinkgoT())
-		namespace, nsCleanupFunc = newNamespace(GinkgoT(), c, genName("webhook-test-"))
+		nsLabels = map[string]string{
+			"foo": "bar",
+		}
+		namespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   genName("webhook-test-"),
+				Labels: nsLabels,
+			},
+		}
+
+		var err error
+		namespace, err = c.KubernetesInterface().CoreV1().Namespaces().Create(context.TODO(), namespace, metav1.CreateOptions{})
+		Expect(err).Should(BeNil())
+		Expect(namespace).ShouldNot(BeNil())
+
+		nsCleanupFunc = func() {
+			err := c.KubernetesInterface().CoreV1().Namespaces().Delete(context.TODO(), namespace.GetName(), metav1.DeleteOptions{})
+			Expect(err).Should(BeNil())
+		}
 	})
 	AfterEach(func() {
 		if nsCleanupFunc != nil {
```

Comment:
should this ever be nil?

Reply:
It should not - but it prevents a potential nil pointer problem. It might be worth changing the if statement to `Expect(nsCleanupFunc).ShouldNot(BeNil())`